src/LibraryViewExtensionWebView2/web/library/library.html

```diff
@@ -155,16 +154,13 @@
                      continue;
                 }
                 //request the icon from the extension.
-                await window.chrome.webview.hostObjects.bridgeTwoWay.GetBase64StringFromPath(src).sync();
+                var base64String = await window.chrome.webview.hostObjects.bridgeTwoWay.GetBase64StringFromPath(src);
```

Comment:
if GetBase64StringFromPath() method is not async, should we still use await?

Reply:
my understanding from reading this page:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.addhostobjecttoscript?view=webview2-dotnet-1.0.1418.22

is that even if the method is not async - the proxy wrapper that webView2 generates to access it returns a promise and needs to be awaited or resolved using `then()` - do you understand these docs differently?

Reply:
yes, you are right when using AddHostObjectToScript you have to wait a promise. Thanks